### PR TITLE
Feature/carthage support

### PR DESCRIPTION
**Pull request by lordkev**
Adds Carthage support per discussion in https://github.com/couchbase/couchbase-lite-ios/issues/928

### CODE REVIEW

File: CouchbaseLite.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3517,7 +3517,6 @@
 					};
 					27452BB71AB1180800327776 = {
 						CreatedOnToolsVersion = 6.3;
-						DevelopmentTeam = N2Q372V7W2;
```
Comment (tleyden) [inline CouchbaseLite.xcodeproj/project.pbxproj]: @lordkev was this meant to be in the PR?

Comment (tleyden) [inline CouchbaseLite.xcodeproj/project.pbxproj]: @lordkev hmm, on further thought, maybe this is just xcode moving stuff around ...

**Comment by tleyden** [comment]
I tried it the PR branch with no luck:

```
$ cat Cartfile
github "buzzfeed/couchbase-lite-ios"
```

```
$ carthage update
*** Cloning couchbase-lite-ios
*** Checking out couchbase-lite-ios at "v1.495"
*** xcodebuild output can be found in /var/folders/6_/n30fk4vj36dggx3tt7m7cnxw0000gp/T/carthage-xcodebuild.1JyUNY.log
*** Skipped building couchbase-lite-ios due to the error:
Dependency "couchbase-lite-ios" has no shared framework schemes

If you believe this to be an error, please file an issue with the maintainers at https://github.com/buzzfeed/couchbase-lite-ios/issues/new
```

**Comment by tleyden** [comment]
Using the commit  hash,  I managed to get further:

```
$ cat Cartfile
github "buzzfeed/couchbase-lite-ios" "b6bd74c5f4f987fb9df4ec5163496a030b4d9106"
```

```
$ carthage update
*** Fetching couchbase-lite-ios
*** Checking out couchbase-lite-ios at "b6bd74c5f4f987fb9df4ec5163496a030b4d9106"
*** xcodebuild output can be found in /var/folders/6_/n30fk4vj36dggx3tt7m7cnxw0000gp/T/carthage-xcodebuild.hAOudT.log
*** Building scheme "CBL iOS framework - Carthage" in CouchbaseLite.xcodeproj
*** WARNING: Skipping build for "iphonesimulator" SDK because the necessary signing identity "iPhone Developer" is not installed
*** WARNING: Skipping build for "iphoneos" SDK because the necessary signing identity "iPhone Developer" is not installed
*** Building scheme "CBL Mac" in CouchbaseLite.xcodeproj
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
ld: warning: ignoring undefined symbol .objc_class_name_CBLUITableSource from -exported_symbols_list
ld: warning: ignoring undefined symbol .objc_class_name_CBLUITableSource from -exported_symbols_list
A shell task failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)

```

**Comment by lordkev** [comment]
I'm out at the moment, but try using --platform ios. Otherwise it'll try
building for all platforms.

**Comment by tleyden** [comment]
Thanks, that avoided the build error, but I'm still getting these code signing identity issues:

```
$ carthage update --platform ios
*** Fetching couchbase-lite-ios
*** Checking out couchbase-lite-ios at "b6bd74c5f4f987fb9df4ec5163496a030b4d9106"
*** xcodebuild output can be found in /var/folders/6_/n30fk4vj36dggx3tt7m7cnxw0000gp/T/carthage-xcodebuild.A76ZBH.log
*** Building scheme "CBL iOS framework - Carthage" in CouchbaseLite.xcodeproj
*** WARNING: Skipping build for "iphonesimulator" SDK because the necessary signing identity "iPhone Developer" is not installed
*** WARNING: Skipping build for "iphoneos" SDK because the necessary signing identity "iPhone Developer" is not installed
```

I'm able to build couchbase-lite-ios from it's own xcode project without an issue, using the default settings (code signing identity is set to "iPhone Developer")

**Comment by tleyden** [comment]
@pasin any chance you can give this a shot and see if you get the same error?

* `$ brew install carthage`
* create new xcode ios project
* add a new file named `Cartfile` with the contents mentioned in https://github.com/couchbase/couchbase-lite-ios/pull/994#issuecomment-164286710
* `$ carthage update --platform ios`

**Comment by pasin** [comment]
Never used carthage before but Let me try that.

**Comment by pasin** [comment]
@tleyden I was able to run `carthage update --platform ios` without the code signing errors. The framework was successfully build in `Carthage/Build/iOS`. There are some warning at the end but I think they are ignorable.

I'm using the latest XCode 7.2. I have an apple developer account setup in XCode > Preferences > Acccount.

```
warning: no rule to process version file '/Users/pasin/Library/Developer/Xcode/DerivedData/CouchbaseLite-cgsqjylstnpipdgfdkfkrnfeluxw/Build/Intermediates/CouchbaseLite.build/Release-iphoneos/CBL iOS framework.build/DerivedSources' for architecture armv7
warning: no rule to process version file '/Users/pasin/Library/Developer/Xcode/DerivedData/CouchbaseLite-cgsqjylstnpipdgfdkfkrnfeluxw/Build/Intermediates/CouchbaseLite.build/Release-iphoneos/CBL iOS framework.build/DerivedSources' for architecture arm64
warning: no rule to process version file '/Users/pasin/Library/Developer/Xcode/DerivedData/CouchbaseLite-cgsqjylstnpipdgfdkfkrnfeluxw/Build/Intermediates/CouchbaseLite.build/Release-iphonesimulator/CBL iOS framework.build/DerivedSources' for architecture i386
warning: no rule to process version file '/Users/pasin/Library/Developer/Xcode/DerivedData/CouchbaseLite-cgsqjylstnpipdgfdkfkrnfeluxw/Build/Intermediates/CouchbaseLite.build/Release-iphonesimulator/CBL iOS framework.build/DerivedSources' for architecture x86_64
```

**Comment by tleyden** [comment]
@snej I think this is safe to merge, given that it worked for @pasin

File: CouchbaseLite.xcodeproj/project.pbxproj (hunk 1 above)
Comment (lordkev) [inline CouchbaseLite.xcodeproj/project.pbxproj]: Yeah, it was meant to be there. If the dev team is specifically set to your Couchbase team it seems to cause issues.

**Comment by jschmid** [comment]
Any update on this ?

**Pull request merged**